src/lit-element.ts

```diff
@@ -288,7 +294,28 @@ export class LitElement extends UpdatingElement {
   }
 
   connectedCallback() {
-    super.connectedCallback();
+    let shouldEnable = true;
+    // If element is pending hydration, in a shadow root, and it's host has not
+    // yet updated, add self to host's `_childrenPendingHydration` list and wait
+    // for host to enable
+    if (this._needsHydration) {
+      const root = this.getRootNode();
+      if (root instanceof ShadowRoot) {
+        // This assumes hydration host is always another LitElement (or
+        // something) implementing the `_childrenPendingHydration` protocol
+        const host = root.host as LitElement;
+        if (!host.hasUpdated) {
```

Comment:
`hasUpdated` is protected, right? Let's just change that to public.

Reply:
Why? This isn't an access violation since it's from a subclass (even if it's on a different instance).